File: class2/1874.c
#include<stdio.h>

int main(){
    int N;
    int top = -1;
    scanf("%d",&N);
    int data[N];
    int stack[N];
    int inp[N];

    for(int j =0; j<N;j++){
        data[j] = 0;
        stack[j]=0;
    }

    for(int i = 0; i< N;i++){
        scanf("%d",inp[i]);
        while(1){
            if(top ==-1 || stack[top] < inp[i]){
                printf("+\n");
                
            }
        }
    }

    for

    for(int i =1;i<=N;i++){
        data[i] = i;
    }
}

File: class2/1874.py
N = int(input())
answer = []
index = 0

arr = [0] * N  # 아직 stack에 안들어감
for _ in range(N):
    if index<0:
        index =0
    inp = int(input()) -1   #입력 받은 수를 index화
    if arr[inp] == 0:
        while index != inp :
            if arr[index] == 0:
                answer.append("+")
                arr[index] = 1
                index +=1
            elif arr[index] == -1 or arr[index] == 1:
                index +=1
        answer.append("+")
        answer.append("-")
        arr[index] = -1
        index -=1
    else :
        while index != inp :
            if arr[index] == 1:
                print("NO")
                exit()
            elif arr[index] == -1:
                index -=1
        answer.append("-")
        arr[index] = -1
        index -=1

print(*answer,sep = '\n')